detox/e2e/support/server_api/client.js

```diff
@@ -3,10 +3,7 @@
 
 import axios from 'axios';
 
-import testConfig from '../test_config';
-
 export const client = axios.create({
-    baseURL: testConfig.siteUrl,
```

Comment:
Looks weird to see baseUrl being passed on every time it calls API functions. Maybe set the baseUrl only for new client and recreate for other server?

Reply:
If I did that, I think I'll have to pass in the client object around to the api classes (e.g. Channel) which is similar to just passing the url. Maybe I'm missing something.

Reply:
@saturninoabril  is not weird, required for multi-server support. Also every single action in the new code has as a first argument the serverUrl

Reply:
I agree and it may be well suited on how the "detox/.../server_api/client.js" is laid out currently. I initially thought of moving away with it, and take advantage of using "app/client/rest/index.ts" for Detox where each server could have its own client and access its properties without passing around the baseUrl. There will be convenient functions where client will be passed around as mentioned by Joe but I'd assume it will be less.
Totally non-blocking for this PR but I'd recommend in doing so for gekidou.

Reply:
@saturninoabril I see what you're saying. I'm a little hesitant to go that approach because I believe the set of clients that mobile has (or will have) won't the full set of APIs one might need to do proper test setup. That would lead to a mix and match of API implementation with some using mobile clients and some using detox custom client.

That said, if later down the road, we determine that mobile clients will have broader scope (like in the web), then with the current approach, we can easily re-do the internal implementation of the functions to use the mobile client.

Reply:
Sounds good to me, thanks Joe!